tas-client/src/tas-client/FeatureProvider/TasApiFeatureProvider.ts

```diff
@@ -38,16 +39,48 @@ export class TasApiFeatureProvider extends FilteredFeatureProvider {
         }
 
         //axios webservice call.
-        let response: AxiosResponse<TASFeatureData> = await this.httpClient.get({ headers: headers });
+        let response: AxiosResponse<TASFeatureData> | undefined;
+
+        try {
+            /**
+             * As mentioned in the axios docs:
+             * Axios Promise will handle handle the catch
+             * section of promises whenever a request is not succesful.
+             * https://axios-http.com/docs/handling_errors
+             */
+            response = await this.httpClient.get({ headers: headers });
+        } catch (error) {
+            const axiosError = error as AxiosError;
+            const properties: Map<string, string> = new Map();
+            if (axiosError.response) {
+                // The request was made and the server responded with a status code
+                // that falls out of the range of 2xx
+                properties.set('ErrorType', 'ServerError');
```

Comment:
Nit: maybe factor `'ErrorType'` into its own constant since it's used in three separate places here.

Reply:
done!